### PR TITLE
fixes #46 : Add psr/log v2/3 support

### LINKED ISSUE #46: Add support for Psr/Log v2/3
I recently developped the [commerce_alma](https://www.drupal.org/project/commerce_alma) module for Drupal Commerce. I'd like to make it available for Drupal 10, however, as D10 now requires Psr/Log v3, the Alma API client install fails.

### PR DESCRIPTION
**Pull request by NicolasGraph**
The PR adds psr/log v2/3 but would require PHP 7.1.0.

v3 is notably required for Drupal10+ and v2 is useful for Sylius, as mentioned by @dunglas.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -20,7 +20,7 @@
   ],
   "require": {
     "php": ">=5.6",
-    "psr/log": "^1.1",
+    "psr/log": "^1.1 || ^3.0",
```
[the remark this answers is not in the document]
Comment (NicolasGraph) [reply composer.json:23]: I personally don't need the v2 so I haven't added or tested it but I guess it should work.

[the remark this answers is not in the document]
Comment (dunglas) [reply composer.json:23]: We tested it and it works. This is necessary for Sylius compatibility for instance.

**Review by dunglas — APPROVED** [review]

**Comment by francoisdelage** [comment]
+1 :))

File: src/Client.php [hunk 2]
```diff
@@ -163,7 +163,7 @@ private function initUserAgent()
      *
      * @return void
      */
-    public function setLogger(LoggerInterface $logger)
+    public function setLogger(LoggerInterface $logger): void
```
Comment (jbcr) [inline src/Client.php:166]: This change break the PHP 5.6 compatibility

File: src/ClientContext.php [hunk 3]
```diff
@@ -85,7 +85,7 @@ public function forcedTLSVersion()
      *
      * @return void
      */
-    public function setLogger(LoggerInterface $logger)
+    public function setLogger(LoggerInterface $logger): void
```
Comment (jbcr) [inline src/ClientContext.php:88]: This change break the PHP 5.6 compatibility

**Review by jbcr — CHANGES_REQUESTED** [review]

File: src/Client.php (hunk 2 above)
Comment (NicolasGraph) [reply src/Client.php:166]: Good catch, this requires PHP 7.1.0. I actually didn't expect Alma to support PHP 5.6.
I add a note about this in the related issue.

File: src/ClientContext.php (hunk 3 above)
Comment (NicolasGraph) [reply src/ClientContext.php:88]: Good catch, this requires PHP 7.1.0. I actually didn't expect Alma to support PHP 5.6.
I add a note about this in the related issue.

**Comment by ClaireAlma** [comment]
Hello, we need to keep the support of PHP 5.6 to 8.2

**Pull request closed without merging**